akka-stream/src/main/scala/akka/stream/impl/FanoutProcessor.scala

```diff
@@ -79,11 +79,9 @@ private[akka] abstract class FanoutOutputs(val maxBufferSize: Int, val initialBu
     case SubscribePending ⇒
       subscribePending()
     case RequestMore(subscription, elements) ⇒
-      // FIXME can we avoid this cast?
       moreRequested(subscription.asInstanceOf[ActorSubscriptionWithCursor[Any]], elements)

```

Comment:
yeah, I guess that was my thought as well. You decide, this is how the alternative looks like:

    case RequestMore(subscription: ActorSubscriptionWithCursor[Any], elements) ⇒
      moreRequested(subscription, elements)
      pump.pump()
    case Cancel(subscription: ActorSubscriptionWithCursor[Any]) ⇒
      unregisterSubscription(subscription)
      pump.pump()
    case RequestMore(subscription, _) =>
      throw new IllegalStateException(s"Expected ${classOf[ActorSubscriptionWithCursor[Any]]}, but was " +
        (if (subscription == null) null else subscription.getClass))
    case Cancel(subscription) =>
      throw new IllegalStateException(s"Expected ${classOf[ActorSubscriptionWithCursor[Any]]}, but was " +
        (if (subscription == null) null else subscription.getClass))